### ISSUE TITLE
[事前学習] - NDL 書籍データを用いた Long-context 性能の改善

### ISSUE THREAD

**Issue by TONJIRU1** [llm-jp/experiments#451]
## Overview

LLM-jp v4 2B モデルを NDL（国立国会図書館）OCR フィルタ済みデータで継続事前学習（Continual Pre-training）する。フィルタリング済み NDL コーパスを v4 トークナイザーで処理したデータを用い、既存チェックポイント（iter_1500000）から +65,516 ステップ追加学習することで、NDL ドメイン適応モデルを構築する。

## Details

ベースモデル `0376_v4-2b-candidate2-pre-phase1`（iter_1500000、約 3.15T tokens 学習済み）から継続学習を行う。

**学習設定**

- スケジューラ: WSD（`-override-opt_param-scheduler`、minus_sqrt decay）
- 追加ステップ数: 65,516 steps（iter_1500000 → iter_1565516）
- グローバルバッチサイズ: 512（シーケンス長 4096）
- チェックポイント保存間隔: 1,000 steps

**学習データ**

| データ | トークン数 |
| --- | --- |
| ndlocrpart_output（NDL OCR） | 4,507,107,400 |
| linepart_alignment_output（NDL 整列） | 6,873,987,538 |

## Resources

**計算機**

- クラスタ: mdx（llm-jp クラスタ）共通 GPU ノード
- ノード種別: GPU（NVIDIA A100-SXM4-40GB × 8）
- ノード台数: 4（32 GPU）

**モデル**

- ベースチェックポイント: `0376_v4-2b-candidate2-pre-phase1/tasks/base/checkpoints/iter_1500000`
- トークナイザー: v4_alpha_1.0.model（vocab: 196,608）

**コード**

- リポジトリ: FIXME [https://github.com/{org}/{repo}](https://github.com/%7Borg%7D/%7Brepo%7D)
- コミット: FIXME xxxxxx
- フレームワーク: Megatron-LM（branch: llmjp0-mdx）

**入力データ**

- MDX: `/model/fujita/ndl_tokenized_v4/`
- ABCI: `/groups/gcg51557/experiments/0248_ndl/data/ndl_tokenized_v4_clean/`

**出力データ**

- 保存先: FIXME
- チェックポイント形式: torch_dist

**W&B ログ**

なし（`WANDB_MODE=offline`）

## 開始日

2026-09-10

## 終了予定日

2026-12-10
